app/components/check_your_answers_summary_component.rb

```diff
@@ -10,11 +10,70 @@ def initialize(model:, title:, fields:, delete_link_to: nil)
   attr_reader :title
 
   def rows
-    fields.map { |key, opts| opts ? row_for_field(key, opts) : nil }.compact
+    fields_with_translations.filter_map do |field|
+      field.options ? row_for_field(field.key, field.options) : nil
+    end
   end
 
   private
 
+  class Field
+    attr_reader :key, :options
+
+    def initialize(field_array)
+      @key = field_array[0]
+      @options = field_array[1]
+    end
+  end
+
+  def fields_with_translations
+    fields_as_array.dup.tap do |fields_to_return|
+      translatable_field_indexes.reverse.each do |index|
+        translatable_item = fields_as_array[index]
+
+        fields_to_return.insert(
+          index + 1,
+          translation_row_for(translatable_item)
+        )
+      end
+    end
```

Comment:
```suggestion
    fields_as_array.flat_map do |field|
      if has_translation?(field)
        [field, translation_row_for(field)]
      else
        [field]
      end
    end
```

I think this makes the code easier to read, and avoids the need for `translatable_field_indexes`.